frontend/src/components/docs/InstructionDocs.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { ReactNode } from 'react';
+import { Typography } from '@material-ui/core';
+import Link from 'components/common/Link';
+import DocsSection from './DocsSection';
+
+type ArgType = 'VAL' | 'REG' | 'STACK';
+
+interface InstructionReference {
+  name: string;
+  args: ArgType[];
+  summary: ReactNode;
+  moreInfo?: ReactNode;
+  errorCases?: ReactNode[];
+  notes?: ReactNode[];
+  examples: string[];
+}
+
+const INSTRUCTIONS: InstructionReference[] = [
+  {
+    name: 'READ',
+    args: ['REG'],
+    summary: (
+      <>
+        Read the next value from <code>INPUT</code> and store it in a register.
+      </>
+    ),
+    moreInfo: (
+      <>
+        The value is removed from <code>INPUT</code>, and cannot be returned.
+      </>
+    ),
+    errorCases: [
+      <>
+        Reading while <code>INPUT</code> is empty causes a runtime error.
+      </>,
+    ],
+    examples: ['READ RX0 ; Move the first value in INPUT into RX0'],
+  },
+  {
+    name: 'WRITE',
+    args: ['VAL'],
+    summary: (
+      <>
+        Write a value to <code>OUTPUT</code>.
+      </>
+    ),
+    moreInfo: (
+      <>
+        Once a value has been written to <code>OUTPUT</code>, it cannot be
+        changed, moved, or removed. The source value is not modified.
+      </>
+    ),
+    examples: [
+      'WRITE 3   ; Write 3 to OUTPUT',
+      'WRITE RX0 ; Write the value in RX0 to OUTPUT',
+    ],
+  },
+  {
+    name: 'SET',
+    summary: 'Set a register to a value.',
+    args: ['REG', 'VAL'],
+    examples: [
+      'SET RX0 3   ; RX0 now holds the value 3',
+      'SET RX0 RX1 ; RX0 now holds whatever value is in RX1',
+    ],
+  },
+  {
+    name: 'ADD',
+    summary: 'Add a value to a register.',
+    moreInfo: (
+      <>
+        The result is stored in the register. See{' '}
+        <Link to="#values--overflow-and-underflow">
+          arithmetic overflow and underflow
+        </Link>
```

Comment:
This is mostly me thinking out load and not a big deal but you think we could make it fill in the link text dynamically based on the id we link to? That way if we want to change the title or w/e we wouldn't need to update all the links